### PR TITLE
182 generalise parsing methods for reading in pheval result tsv files

### PR DESCRIPTION
**Pull request by yaseminbridges**
Originally there were 3 separate methods for parsing the tsv result to a pheval gene, variant or disease result. I have now generalised this into a single method and instead allow the specification of what result type this needs to be parsed into and moved into its own file separate to the analysis methods

### CODE REVIEW

File: src/pheval/analyse/analysis.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# #!/usr/bin/python
```
Comment (matentzn) [inline src/pheval/analyse/analysis.py:1]: Why was this here in the first place?

File: src/pheval/analyse/analysis.py [hunk 2]
```diff
@@ -16,6 +15,7 @@
     generate_benchmark_gene_output,
     generate_benchmark_variant_output,
 )
+from pheval.analyse.parse_pheval_result import _read_standardised_result, parse_pheval_result
```
Comment (matentzn) [inline src/pheval/analyse/analysis.py:18]: Is it a good idea to import a `private` `_...` method to another file?

File: src/pheval/analyse/parse_pheval_result.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from pathlib import Path
+
+import pandas as pd
+
+from pheval.post_processing.post_processing import PhEvalResult
+
+
+def _read_standardised_result(standardised_result_path: Path) -> dict:
```
Comment (matentzn) [inline src/pheval/analyse/parse_pheval_result.py:8]: This does not seem to be a private method so why the underscore?

**Review by matentzn — COMMENTED** [review]
Looks great, just a tiny suggestion!

File: src/pheval/analyse/analysis.py (hunk 1 above)
Comment (yaseminbridges) [reply src/pheval/analyse/analysis.py:1]: Definitely some type of mistake on my end, very likely to have happened at the beginning of the project

**Review by matentzn — APPROVED** [review]
Great!!

**Review by julesjacobsen — APPROVED** [review]
Nice
